src/ng/compile.js

```diff
@@ -2555,10 +2574,10 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
-            }
             parentGet = $parse(attrs[attrName]);
+
+            if (parentGet === noop && optional) break;
```

Comment:
In the case of `attrs['watch']` is a function, it would been set to `undefined` in line 2551. So $parse(undefined) would be not valid.